spec/analyzer_spec.rb

```diff
@@ -0,0 +1,106 @@
+require 'spec_helper'
+
+RSpec.describe Analyzer do
+    describe '#initialize' do
+        it 'exists' do
+            analyzer = Analyzer.new
+
+            expect(analyzer).to be_a(Analyzer)
+        end
+    end
+
+    describe '#most_popular_vehicle(vehicles)' do
+        it 'can analyze an array of one vehicle and return the make and model' do
+            cruz = Vehicle.new({vin: '123456789abcdefgh', year: 2012, make: 'Chevrolet', model: 'Cruz', engine: :ice})
+            analyzer = Analyzer.new
+
+            vehicles = [cruz]
+
+            expect(analyzer.most_popular_vehicle(vehicles)).to eq(['Chevrolet', 'Cruz'])
```

Comment:
We'd want a more robust test to make sure this method actually figures out the most popular vehicle. Having just one vehicle in the array doesn't give me confidence that this method can actually find the right value when there are 50 items in the array.